models/commit_status.go

```diff
@@ -49,21 +49,21 @@ func init() {
 }
 
 // upsertCommitStatusIndex the function will not return until it acquires the lock or receives an error.
-func upsertCommitStatusIndex(e db.Engine, repoID int64, sha string) (err error) {
+func upsertCommitStatusIndex(ctx context.Context, repoID int64, sha string) (err error) {
 	// An atomic UPSERT operation (INSERT/UPDATE) is the only operation
 	// that ensures that the key is actually locked.
 	switch {
 	case setting.Database.UseSQLite3 || setting.Database.UsePostgreSQL:
-		_, err = e.Exec("INSERT INTO `commit_status_index` (repo_id, sha, max_index) "+
+		_, err = db.Exec(ctx, "INSERT INTO `commit_status_index` (repo_id, sha, max_index) "+
 			"VALUES (?,?,1) ON CONFLICT (repo_id,sha) DO UPDATE SET max_index = `commit_status_index`.max_index+1",
 			repoID, sha)
 	case setting.Database.UseMySQL:
-		_, err = e.Exec("INSERT INTO `commit_status_index` (repo_id, sha, max_index) "+
+		_, err = db.Exec(ctx, "INSERT INTO `commit_status_index` (repo_id, sha, max_index) "+
 			"VALUES (?,?,1) ON DUPLICATE KEY UPDATE max_index = max_index+1",
 			repoID, sha)
 	case setting.Database.UseMSSQL:
 		// https://weblogs.sqlteam.com/dang/2009/01/31/upsert-race-condition-with-merge/
-		_, err = e.Exec("MERGE `commit_status_index` WITH (HOLDLOCK) as target "+
+		_, err = db.Exec(ctx, "MERGE `commit_status_index` WITH (HOLDLOCK) as target "+
```

Comment:
```suggestion
		_, err = db.GetEngine(ctx).Exec("MERGE `commit_status_index` WITH (HOLDLOCK) as target "+
```
why not this :point_up: 

what are the benefits to recreate the interface (+context added) of engine with the db package?

Reply:
It's shorter. :)